metafix/src/main/java/org/metafacture/metafix/Record.java

```diff
@@ -153,4 +190,52 @@ public void retainFields(final Collection<String> fields) {
         super.retainFields(fields);
     }
 
+    /**
+     * Transform this record by applying the given operator to all matching values for the given field.
+     *
+     * @param field The field
+     * @param operator The operator
+     */
+    public void transform(final String field, final UnaryOperator<String> operator) {
+        final FixPath findPath = new FixPath(field);
+        final Value found = findPath.findIn(this);
+        findPath.throwIfNonString(found);
+        Value.asList(found, results -> {
+            for (int i = 0; i < results.size(); ++i) {
```

Comment:
You can't iterate in ascending order if you're going to [remove](https://github.com/metafacture/metafacture-fix/pull/170#discussion_r823631287) elements based on the index number - they're going to be off once the first element has been removed.

Reply:
Here's a test for the [above mentioned](https://github.com/metafacture/metafacture-fix/pull/170#issuecomment-1081983278) error (introduced by f210dd3): a816a73. It's exactly the issue described here.

Reply:
Right, thanks for the details and test case. Fixed with ec9f8ef.